client/client.go

```diff
@@ -64,17 +68,20 @@ func forgetSrv(name string) {
 	srvLookupMutex.Unlock()
 }
 
-// randomBroker tries to resolve name through a call to lookupSrv. If
-// successful it returns a random host:port from the list. If lookupSrv fails
-// it returns name unmodified (so you can pass "localhost:9092" for example).
+// randomBroker tries to resolve name through a call to lookupSrv. If successful
+// it returns a random host:port from the list. lookupSrv in its turn invokes
+// net.LookupSRV(), unsuccessful result is considered as not an SRV record, so
+// you can pass "localhost:9092" for example. It panics, if resolved SRV record
+// has no hosts.
 func randomBroker(name string) string {
 	addrs, err := lookupSrv(name)
 	if err != nil {
-		return name
-	}
-	if len(addrs) == 0 { // is this possible?
-		return name
+		if errors.Is(err, errNotAnSRV) {
+			return name
+		}
+		panic(err)
```

Comment:
Returning an error and handling it up the chain would be better.

Reply:
this is what I decided to do when I woke up today.